cni-plugin/tests/calico_cni_k8s_test.go

```diff
@@ -3099,7 +3105,7 @@ var _ = Describe("Kubernetes CNI tests", func() {
 			clientset := getKubernetesClient()
 
 			// Now create a K8s pod.
-			name := "mypod-1"
+			name := fmt.Sprintf("pod-%d", rand.Uint32())
```

Comment:
1. Make a utility function to encapsulate the pod name generation?
2. I suppose it's astronomically unlikely, but in theory we can still get duplicate names with this approach.  Did you consider something like an incrementing index?